lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,7 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        raise Puppet::Error.new( "Puppet is not able to create or delete Active Directory users with the user resource", e ) if /8007089A in Active Directory/ =~ e.message
```

Comment:
The trouble is that `DOMAIN` could be the local computer, `.`, or `NT AUTHORITY` I think, and those we do support. You might be able to resolve the name to a SID, and ask the object if it's a built-in account or not. If not, check the domain and against the list of well known domains... but that gets a bit kludgy...